packages/lib/shim-init-node.js

```diff
@@ -420,11 +436,18 @@ function shimInit(options = null) {
 		return new Buffer(data).toString('base64');
 	};
 
-	shim.fetch = async function(url, options = null) {
+	shim.fetch = async function(url, options = {}) {
 		const validatedUrl = urlValidator.isUri(url);
 		if (!validatedUrl) throw new Error(`Not a valid URL: ${url}`);
+		const proxyUrl = fetchProxyUrl();
+		options.agent = proxyUrl ? shim.proxyAgent(url, proxyUrl) : null;
+
+		return shim.fetchWithRetry((requestWithProxy) => {
+			// If a proxy has been set but it failed, try again without proxy agent
+			if (options?.agent?.proxy && requestWithProxy === false) {
+				options.agent = null;
+			}
```

Comment:
Hmm, I don't think we should have this, that's going to be too confusing. If it fails, it fails, and the user will need to check their proxy.

Reply:
It’s a nice quality of life addition for only a few lines

Reply:
But it feels like it's going to be confusing. For example, what if someone has a hard requirement to use a proxy? Then randomly there would be non-proxy calls. It also means the user won't even know their proxy is not being used or doesn't work. At least for a first version I would prefer this out, and if there's a need for it at some point it can be discussed in a new issue.

Reply:
That’s fair enough